### PR TITLE
[Serialization] Implement serialization and deserialization

### PR DESCRIPTION
**Pull request by Buga1chukY**

### CODE REVIEW

File: src/main/java/serialization/Person.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String toString() {
```
Comment (MLGMAG) [inline src/main/java/serialization/Person.java:22]: Try to use generated `toString` method.
Press `ALT+Insert` shortcut and generate it

File: src/main/java/serialization/ReadObject.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+
+public class ReadObject {
+    public static void main(String[] args) {
+
+        try {
+            FileInputStream fis = new FileInputStream("people.bin");
```
Comment (MLGMAG) [inline src/main/java/serialization/ReadObject.java:11]: You forget to deallocate the descriptor

File: src/main/java/serialization/WriteObject.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+public class WriteObject {
+    public static void main(String[] args) {
+        Person person1 = new Person(1, "Bob");
+        Person person2 = new Person(2, "Mike");
+
+        try {
+            FileOutputStream fos = new FileOutputStream("people.bin");
```
Comment (MLGMAG) [inline src/main/java/serialization/WriteObject.java:13]: You forget to deallocate the descriptor

File: src/main/java/serialization/ReadObject.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            Person person2 = (Person) ois.readObject();
+
+            System.out.println(person1);
+            System.out.println(person2);
+
+            ois.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
```
Comment (MLGMAG) [inline src/main/java/serialization/ReadObject.java:24]: Check the best practices to handle the exceptions in Java.

File: people.bin
Comment (MLGMAG) [inline people.bin:1]: Remove this file, it is no a source code.

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/serialization/ReadObject.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            System.out.println("Class not found");
+            e.printStackTrace();
+        } catch (IOException e) {
+            System.out.println("IO exception");
+            e.printStackTrace();
+        }
+    }
+}
```
Comment (MLGMAG) [inline src/main/java/serialization/ReadObject.java:27]: Always add new line at the end of the file

File: src/main/java/serialization/ReadObject.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package main.java.serialization;
+
+import java.io.*;
+
+public class ReadObject {
+    public static void main(String[] args) {
+
+        try (FileInputStream fis = new FileInputStream("people.bin");
```
Comment (MLGMAG) [inline src/main/java/serialization/ReadObject.java:8]: Extract `people.bin` into the constant

File: src/main/java/serialization/ReadObject.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            System.out.println(person2);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            System.out.println("Class not found");
+            e.printStackTrace();
+        } catch (IOException e) {
+            System.out.println("IO exception");
+            e.printStackTrace();
```
Comment (MLGMAG) [inline src/main/java/serialization/ReadObject.java:24]: What is the difference between `printStackTrace` and `getMessage`?

File: src/main/java/serialization/WriteObject.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+public class WriteObject {
+    public static void main(String[] args) {
+        Person person1 = new Person(1, "Bob");
+        Person person2 = new Person(2, "Mike");
+
+        try (FileOutputStream fos = new FileOutputStream("people.bin");
```
Comment (MLGMAG) [inline src/main/java/serialization/WriteObject.java:12]: Extract `people.bin` into the constant

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/serialization/ReadObject.java (hunk 7 above)
Comment (Buga1chukY) [reply src/main/java/serialization/ReadObject.java:24]: photo_2023-10-10_15-37-41  Method printStackTrace give us more information about error, we recive the stack call when exception was thrown.


photo_2023-10-10_15-37-33 GetMessage gie us reason of the error. We can understand why it was happend, but we don't know in what time if code it happend.

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**